src/Google.Bigquery.V2/PollJobOptions.cs

```diff
@@ -0,0 +1,85 @@
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Google.Bigquery.V2
+{
+    /// <summary>
+    /// Options for <c>PollJob</c> operations. When no options are specified, the
+    /// default behavior is to poll forever, at one second intervals.
+    /// </summary>
+    public sealed class PollJobOptions
+    {
+        private TimeSpan? _interval;
+
+        /// <summary>
+        /// The interval to delay between poll 
+        /// requests.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> is negative.</exception>
+        public TimeSpan? Interval
+        {
+            get { return _interval; }
+            set
+            {
+                if (value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(Interval)} cannot be negative");
+                }
+                _interval = value;
+            }
+        }
+
+        /// <summary>
+        /// The maximum number of poll requests to make.
+        /// </summary>
+        /// <remarks>
+        /// If the job has not completed after this number of requests has been made,
+        /// the poll operation will fail with a <see cref="TimeoutException"/> as if
+        /// a timeout had been set.
+        /// </remarks>
+        public int? MaxRequests { get; set; }
+
+        /// <summary>
+        /// The maximum overall time to poll for. Populating this property
+        /// is equivalent to setting the <see cref="Deadline"/> property
+        /// to "the time of calling <c>PollJob</c>, plus the timeout".
+        /// </summary>
+        /// <remarks>
+        /// A <see cref="PollJobOptions"/> with both <see cref="Timeout"/>
+        /// and <see cref="Deadline"/> set to non-null values is invalid,
+        /// and will cause an exception to be thrown.
+        /// </remarks>
+        public TimeSpan? Timeout { get; set; }
+
+        /// <summary>
+        /// A deadline for polling; no requests will be made after this point
+        /// in time.
+        /// </summary>
+        /// <remarks>
+        /// Currently, requests are not cancelled at this deadline; it only
+        /// affects when the last request is started.
+        /// </remarks>
+        public DateTimeOffset? Deadline { get; set; }
+
+        internal void Validate()
+        {
+            Preconditions.CheckArgument(Timeout == null || Deadline == null, "options",
+                $"Cannot set both {nameof(Timeout)} and {nameof(Deadline)} to non-null values");
+        }
+
+        internal DateTimeOffset? GetEffectiveDeadline() => Deadline ?? DateTime.UtcNow + Timeout;
```

Comment:
Indeed - it would be weird to pass around, and I don't think it's worth it just for this.